modal/partial_function.py

```diff
@@ -394,12 +406,24 @@ def create_wsgi() -> Callable:
     """
     if isinstance(_warn_parentheses_missing, str):
         raise InvalidError('Positional arguments are not allowed. Suggestion: `@wsgi_app(label="foo")`.')
-    elif _warn_parentheses_missing:
+    elif _warn_parentheses_missing is not None:
         raise InvalidError(
             "Positional arguments are not allowed. Did you forget parentheses? Suggestion: `@wsgi_app()`."
         )
 
     def wrapper(raw_f: Callable[..., Any]) -> _PartialFunction:
+        if callable_has_non_self_params(raw_f):
+            if callable_has_non_self_non_default_params(raw_f):
+                raise InvalidError(
+                    f"WSGI app function {raw_f.__name__} can't have arguments. See https://modal.com/docs/guide/webhooks#wsgi."
```

Comment:
Not sure if this is idiosyncratic or pointless, but I have always thought (in Python at least) that *argument* refers to the value passed into the function while *parameter* refers to the placeholder for that value in the function definition.

Reply:
Valid point, updated the language to be more precise.